src/main/java/bio/terra/common/db/DatabaseRetryUtils.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.common.db;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.RecoverableDataAccessException;
+import org.springframework.dao.TransientDataAccessException;
+
+/** Utilities to execute database operations with retry support. */
+public class DatabaseRetryUtils {
+  private static final Logger logger = LoggerFactory.getLogger(DatabaseRetryUtils.class);
+
+  /** Executes a database operation and retries if retryable. */
+  public static <T> T executeAndRetry(
+      DatabaseExecute<T> execute, Duration retrySleep, int maxNumRetries)
+      throws InterruptedException {
+    int numRetries = 0;
```

Comment:
If you switch to "attempts" semantics for the parameter, it might make sense to switch to attempts here too for consistency, e.g. `int numAttempts = 1;`